packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -352,6 +355,44 @@ class TransactionService {
     )
   }
 
+  setSession(tr) {
+    let session = this._config.get('session')
+
+    if (session) {
+      if (typeof session == 'boolean') {
+        tr.session = {
+          id: generateRandomId(16),
+          sequence: 1
+        }
+      } else {
+        if (
+          session.timestamp &&
+          Date.now() - session.timestamp > SESSION_TIMEOUT
+        ) {
+          tr.session = {
+            id: generateRandomId(16),
+            sequence: 1
+          }
+        } else {
+          tr.session = {
+            id: session.id,
+            sequence: session.sequence ? session.sequence + 1 : 1
+          }
+        }
+      }
+
+      const sessionConfig = {
+        session: {
+          id: tr.session.id,
+          sequence: tr.session.sequence,
+          timestamp: Date.now()
```

Comment:
we are updating the timestamp every time session is set which feels incorrect? this should be only happen when we start session for first time or session expired right?

Reply:
As we discussed yesterday, this is supposed to be `lastActive`, in other words, we keep the session active until 30 mins of inactivity. This was what we discussed previously, but can we can change in the future if needed.

Reply:
@vigneshshanmugam , I'm going to keep the name `timestamp` for this field for now, until we finalise the last activity vs start/end timestamps. But this is still useful for testing data transforms, so we can release as is.